packages/manager/modules/pci/src/projects/project/notebooks/notebooks.controller.js

```diff
@@ -0,0 +1,100 @@
+import capitalize from 'lodash/capitalize';
+import get from 'lodash/get';
+
+import { NOTEBOOK_STATUS } from './Notebook.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, CucCloudMessage, CucRegionService, NotebookService) {
+    this.$translate = $translate;
+    this.capitalize = capitalize;
+    this.CucCloudMessage = CucCloudMessage;
+    this.CucRegionService = CucRegionService;
+    this.NotebookService = NotebookService;
+  }
+
+  $onInit() {
+    this.messageContainer = 'pci.projects.project.notebooks';
+    this.loadMessages();
+    this.trackNotebooks('table', 'page');
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(this.messageContainer);
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      this.messageContainer,
+      {
+        onMessage: () => this.refreshMessages(),
+      },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  createNotebook() {
+    this.trackNotebooks('table::create_notebook');
+    this.goToAddNotebook();
+  }
+
+  goToNotebookDetails(notebook) {
+    this.trackNotebooks('table::options_menu::details');
+    this.goToNotebook(notebook);
+  }
+
+  deleteNotebook(notebook) {
+    this.trackNotebooks('table::options_menu::delete_notebook');
+    this.goToDeleteNotebook(notebook);
+  }
+
+  startNotebook(notebook) {
+    this.trackNotebooks('table::options_menu::start_notebook');
+    this.CucCloudMessage.flushMessages(this.messageContainer);
+    this.NotebookService.startNotebook(this.projectId, notebook.id).then(
+      () => {
+        notebook.setState(NOTEBOOK_STATUS.STARTING);
+        this.CucCloudMessage.success(
+          this.$translate.instant('pci_notebook_list_start_success', {
+            notebookName: notebook.name,
+          }),
+          this.messageContainer,
+        );
+      },
+      (error) => {
+        this.CucCloudMessage.error(
+          this.$translate.instant('pci_notebook_list_start_error', {
+            notebookName: notebook.name,
+            message: get(error, 'data.message'),
+          }),
+          this.messageContainer,
+        );
```

Comment:
looks like you are using same code block in multiple times. Can you extract to a separate function?

Reply:
Its just a single function call. Even if extracted to a separate method, it still has to take the same arguments (message and container)...